main/src/main/scala/sbt/Defaults.scala

```diff
@@ -772,8 +773,23 @@ object Defaults extends BuildCommon {
     else Vector()
   }
 
-  def makeCrossTarget(t: File, sv: String, sbtv: String, plugin: Boolean, cross: Boolean): File = {
-    val scalaBase = if (cross) t / ("scala-" + sv) else t
+  @deprecated("Use constructor with scalaVersion and scalaBinaryVersion", "1.5.0")
+  def makeCrossTarget(t: File, bv: String, sbtv: String, plugin: Boolean, cross: Boolean): File = {
+    val scalaBase = if (cross) t / ("scala-" + bv) else t
+    if (plugin) scalaBase / ("sbt-" + sbtv) else scalaBase
+  }
+
+  def makeCrossTarget(
+      t: File,
+      scalaVersion: String,
+      scalaBinaryVersion: String,
+      sbtv: String,
+      plugin: Boolean,
+      cross: Boolean
+  ): File = {
+    val scalaSuffix =
+      if (ScalaArtifacts.isScala3(scalaVersion)) scalaVersion else scalaBinaryVersion
```

Comment:
should the same be used for source directories? lines 745 and 760 above.

also, for lines 745-746 and scala3, is epochVersion different from scalaBinaryVersion?

should the cross-build source be `scala-3`, `scala-3.1` or `scala-3.10`?

Reply:
Since Scala 3 is relatively new, so the best practice around how libraries should be maintained is still in flux, but given that Maven Central artifacts would have `_3` suffix, I don't think library authors would be interested in cross building against the minor releases of Scala 3 like 3.0.x and 3.1.x. Likely they'd either decide to stay on 3.0 or bite the bullet and move to 3.1. So from the usage perspective, supporting source directory `scala-3` and `scala-3.1` out of box doesn't make sense for now in my opinion.

If there are other good reason for this, please open a thread under https://github.com/sbt/sbt/discussions